controllers/auth/auth.controller.ts

```diff
@@ -0,0 +1,134 @@
+import express, { Request, Response, Router } from "express";
+import dotenv from "dotenv";
+import { TwitterApi } from "twitter-api-v2";
+import { ResponseHandler } from "../../services";
+import { usersRef } from "../../db/firebase";
+import { userClient } from "../../TwitterClient";
+import { IOAuth2RequestTokenResult } from "twitter-api-v2/dist/types/auth.types";
+
+const authRouter: Router = express.Router();
+
+// configure env variable
+dotenv.config();
+
+authRouter.get("/", (req: Request, res: Response) => {
+  return res.redirect("/authorize");
+});
+
+authRouter.get("/authorize", async (req: Request, res: Response) => {
+  try {
+    const authLink: IOAuth2RequestTokenResult =
+      userClient.generateOAuth2AuthLink("http://localhost:3000/me", {
+        scope: [
+          "tweet.read",
+          "users.read",
+          "bookmark.read",
+          "bookmark.write",
+          "offline.access",
+        ],
+      });
+    req.session.oAuth = {
+      ...authLink,
+    };
+    ResponseHandler.requestSuccessful({
+      res,
+      payload: { url: req.session.oAuth.url },
+    });
+  } catch (err) {
+    ResponseHandler.serverError(res, "could not generate auth link");
+    console.log("could not generate auth link \n" + err);
+  }
+});
+
+authRouter.get("/me", async (req: Request, res: Response) => {
+  try {
+    if (req.session.oAuth) {
+      // get oauth token and verifier
+      const { state, code } = req.query,
+        // retrieve oath_token_secret from session store
+        { codeVerifier, state: sessionState } = req.session.oAuth;
```

Comment:
I think it's good practice to define assignment to variables separately for readability purposes.
`const something = ...`
`const anotherThing = ...`